render_macros/src/element_attribute.rs

```diff
@@ -94,14 +94,41 @@ impl Hash for ElementAttribute {
 
 impl Parse for ElementAttribute {
     fn parse(input: ParseStream) -> Result<Self> {
-        let name = AttributeKey::parse_separated_nonempty_with(input, syn::Ident::parse_any)?;
-        let not_punned = input.peek(syn::Token![=]);
+        let mut name: syn::punctuated::Punctuated<proc_macro2::Ident, proc_macro2::Punct> =
+            syn::punctuated::Punctuated::new();
+
+        // Parse the input up to the space
+        loop {
+            let value = syn::Ident::parse_any(&input).unwrap();
+            name.push_value(value);
+
+            if input.peek(syn::Token![=]) {
+                break;
+            }
+
+            let punct = input.parse().unwrap();
+            name.push_punct(punct);
+        }
```

Comment:
This loop is similar to the implementation in `parse_separated_nonempty_with`. Key thing I was looking to do here was to give us the ability to modify how and when this accrues / breaks so we can better handle the edge cases. I'm not sure I ended up going very far with this but think it probably makes sense to hold onto this here.